block/block/nft/nember_extract.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+import json
+import pickle
+from binascii import unhexlify
+
+import msgpack
+from tqdm import tqdm
+
+from block.extractor import public_key_to_address
+
+
+def extract_nft_transfer(txn, height, nft_tx_map):
+    for mosaic in txn[b'payload'][b'mosaics']:
+        if mosaic[b'mosaic_id'] in nft_tx_map and mosaic[b'amount'] == 1:
+            nft_tx_map[mosaic[b'mosaic_id']].append((height, txn[b'payload'][b'recipient_address']))
+
+
+def main(args):
+    # pylint: disable=too-many-nested-blocks, too-many-branches
+
+    with open(args.input, 'rb') as file:
+        blocks = msgpack.unpack(file, unicode_errors=None, raw=True)
+
+    nft_mosaics = []
+    platform_tx = []
+    for block in tqdm(blocks):
+        for txn in block[b'footer'][b'transactions']:
+            if txn[b'type'] == b'4241' and txn[b'payload'][b'embedded_tx_count'] == 4:
+                flag = False
+                for sub_tx in txn[b'payload'][b'embedded_transactions']:
+                    if sub_tx[b'type'] == b'4154':
+                        if b'NEMBER.ART' in sub_tx[b'payload'][b'message']:
+                            flag = True
+                if flag:
+                    gen_tx = [x for x in txn[b'payload'][b'embedded_transactions'] if x[b'type'] == b'414d']
+                    meta_tx = [x for x in txn[b'payload'][b'embedded_transactions'] if x[b'type'] == b'4244']
+                    supply_tx = [x for x in txn[b'payload'][b'embedded_transactions'] if x[b'type'] == b'424d']
+                    if any(len(x) != 1 for x in [gen_tx, meta_tx, supply_tx]):
+                        platform_tx.append(txn)
+                        continue
+                    gen_tx = gen_tx[0]
+                    meta_tx = meta_tx[0]
+                    supply_tx = supply_tx[0]
```

Comment:
we usually don't reuse variables like this.
things like `blocks = sorted(blocks)` are fine - does not change the type

this one changes from array to single entity (all 3 gen_tx, meta_tx, supply_tx)